src/api/app/policies/report_policy.rb

```diff
@@ -0,0 +1,34 @@
+class ReportPolicy < ApplicationPolicy
+  attr_reader :user, :record, :reportable
+
+  def initialize(user, record)
+    super(user, record)
+
+    @user = user
+    @record = record
+    @reportable = record.reportable
+  end
+
+  def show?
+    user.is_admin? || record.user == user
+  end
+
+  def create?
+    return false unless Flipper.enabled?(:content_moderation, user)
+
+    # We don't want reports twice...
+    return false if user.submitted_reports.where(reportable: record.reportable).any?
```

Comment:
@danidoni this is not correct. I've fixed this in PR #14880.

Reply:
Ah, I messed up while rebasing. Let me change it